src/validation.cpp

```diff
@@ -5091,19 +5091,22 @@ bool LoadMempool(CTxMemPool& pool)
         }
 
         // TODO: remove this try except in v0.22
+        std::map<uint256, uint256> unbroadcast_txids;
         try {
-          std::set<uint256> unbroadcast_txids;
           file >> unbroadcast_txids;
           unbroadcast = unbroadcast_txids.size();
-
-          for (const auto& txid : unbroadcast_txids) {
-            pool.AddUnbroadcastTx(txid);
-          }
         } catch (const std::exception&) {
           // mempool.dat files created prior to v0.21 will not have an
           // unbroadcast set. No need to log a failure if parsing fails here.
         }
-
+        for (const std::pair<uint256, uint256>& elem : unbroadcast_txids) {
```

Comment:
Believe the same use of `const auto&` is needed here as used in [net_processing](https://github.com/bitcoin/bitcoin/pull/18044/commits/d5d55edb1f295711574190bf310f88171451b5cf#diff-eff7adeaec73a769788bb78858815c91R823).
```suggestion
        for (const auto& elem : unbroadcast_txids) {
```

Reply:
Doh, thanks! Fixed.